src/legacy/ui/public/index_patterns/_index_pattern.js

```diff
@@ -180,6 +180,14 @@ export function IndexPatternProvider(Private, config, Promise, confirmModalPromi
   }
 
   function initFields(indexPattern, input) {
+    if(!indexPattern.fields && Array.isArray(input)) {
```

Comment:
Is there a specific reason why we didn't opt for extending the field format infrastructure to define default formats based on esTypes (basically what @wylieconlon is suggesting down below)?

This would come down to adding a second parameter esTypes to `src/legacy/ui/public/registry/field_formats.js#getDefaultInstance` and extending the defaultTypeMap with the possibility to specify es types.
```
{
  "ip": { "id": "ip", "params": {} },
  "date": { "id": "date", "params": {} },
  "date_nanos": { "id": "date_nanos", "params": {} },
  "number": { "id": "number", "params": {} },
  "boolean": { "id": "boolean", "params": {} },
  "_source": { "id": "_source", "params": {} },
  "_default_": { "id": "string", "params": {} }
}
```

We could look up the esTypes first and then fall back to kibana field types if none are found - which is basically what the current implementation is doing, but without telling the user. cc @timroes

Reply:
I'm implementing this approach, it aligns with @wylieconlon's suggestion, no veto from @timroes so far. We don't need to set a value in fieldFormatMap in the saved object this way, much better.

Reply:
Adaption: I've added an `es` flag, else for instance ES type `boolean` would be used instead Kibana type `boolean` . In practice this makes no difference but theoretically `boolean` could be assigned to a different Kibana Field Type with a different Default Format. Then it would be formatted with the `boolean` format type. It's just to prevent a unlikely future collision.

```
{
  "ip": { "id": "ip", "params": {} },
  "date": { "id": "date", "params": {} },
  "date_nanos": { "id": "date_nanos", "params": {}, "es": true },
  "number": { "id": "number", "params": {} },
  "boolean": { "id": "boolean", "params": {} },
  "_source": { "id": "_source", "params": {} },
  "_default_": { "id": "string", "params": {} }
}
```